application/src/views/Project.vue

```diff
@@ -1,116 +1,183 @@
-        tag: 'Nature',
-        updates: [{
-          date: 'some date',
-          text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.'
-        },
-        {
-          date: 'some date',
-          text: 'some text'
-        },
-        {
-          date: 'some date',
-          text: 'some text'
-        }
-        ]
-      }
+      project_id: this.$route.params.id,
+      subPage: 0,
+      updateToEdit: {},
+      // project : getProject
+      project: {},
+      gravatarUrl
     }
   },
+  async mounted () {
+    this.project = await getProject(this.project_id)
+    this.gravatarUrl = gravatarUrl(this.project.owner.email, 300)
+  },
   methods: {
     toggleDesciptionvisibility (value) {
-      this.descriptionVisible = value
+      this.subPage = value
     },
     timelineSide (index) {
       return (index % 2 === 0) ? 'left' : 'right'
+    },
+    async postUpdate (value) {
+      var date = new Date()
+      var update = await addUpdate(
+        this.project_id,
+        {
+          text: value,
+          author: getUser(),
+          project: this.project,
+          date: date
+        })
+      this.project.updates.push(update)
+
+      this.toggleDesciptionvisibility(1)
+    },
+    setUpdate (update) {
+      this.updateToEdit = update
+      this.toggleDesciptionvisibility(3)
+    },
+    editUpdate (update) {
+      updateUpdate(this.updateToEdit._id, update)
+
+      this.toggleDesciptionvisibility(1)
+    },
+    removeUpdate (id, index) {
+      deleteUpdate(id)
+      this.project.updates.splice(index, 1)
+    },
+    follow () {
+      if (getUser() !== null) {
+        followProject(this.project._id)
+        var user = getUser()
+        user.followedProjects.push((this.project._id))
+        storeUser(user)
+        this.$forceUpdate()
+      } else {
+        window.location = '/login'
+      }
+    },
+    formatDate (date) {
+      var newDate = dayjs(date).format('YYYY/MM/DD')
+      return newDate
+    },
+    editMode () {
+      window.location = '/project/' + this.project._id + '/edit'
+    },
+    apply () {
+      if (getUser() !== null) {
+
+      } else {
+        window.location = '/login'
+      }
     }
   },
   computed: {
-    getProjectData () {
-      // $route.param.id
-      return ''
+    checkMembership () {
+      if (getUser() !== null &&
+      (this.project.members.includes(getUser()._id) ||
+      this.project.owner._id === getUser()._id)) {
+        return true
+      } else {
+        return false
+      }
+    },
+    checkFollowed () {
+      if (getUser() != null) {
```

Comment:
I think this should have another `=`
```suggestion
      if (getUser() !== null) {
```